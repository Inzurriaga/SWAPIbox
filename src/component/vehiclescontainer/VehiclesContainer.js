import React, { Component } from "react"
import VehicleCard from "../vehiclecard/Vehiclecard"
import Loading from "../loading/LoadingScreen"

export default class VehiclesContainer extends Component {
    constructor(){
        super();
        this.state = {
            vehicles: [],
            error: ""
        }
    }

    componentDidMount = async () => {
        if(this.props.vehicleinfo.length){
            console.log("vehicle")
        }else{
            const url = "https://swapi.co/api/vehicles/"
            try {
                const response = await fetch(url)
                const vehicle = await response.json()
                this.setState({
                    vehicles: vehicle.results})
            } catch(error) {
                this.setState({
                    error: error.message
                })
            }
        }
    }

    render(){
        const cards = this.state.vehicles.map((vehicle) => {
            return  <VehicleCard vehicle={vehicle}/>
         })
        return(
            <div className="vehicles-container">
                {
                    this.state.vehicles.length ?
                        cards:
                        <Loading /> 
                }
            </div>
        )
    }
}