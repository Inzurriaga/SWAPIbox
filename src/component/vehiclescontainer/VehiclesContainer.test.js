import React from "react"
import { shallow } from "enzyme"
import VehiclesContainer from "./VehiclesContainer"

const mockVehicleinfo = []
const mockData = []

describe("vehicles_container",() => {
let wrapper;
beforeEach(() => {
    wrapper = shallow(
        <VehiclesContainer vehicleinfo={mockVehicleinfo}/>
    )
    fetch = jest.fn().mockImplementation(() => Promise.resolve({
        status: 200,
        json: () => Promise.resolve(mockData),
    }));
});
it("should match snapshot", () => {
expect(wrapper).toMatchSnapshot()
});
it("should have a default state", () => {
expect(wrapper.state()).toEqual({
    vehicles: [],
    error: ""
})
});
it.skip("should call fetch with the correct url", () => {
const url = "https://swapi.co/api/vehicles/"
wrapper.instance().componentDidMount()
expect(fetch).toHaveBeenCalledWith(url)
});
it("should change state when fetch is resolved", async () => {
expect(wrapper.state("vehicles")).toEqual([])
await wrapper.instance().componentDidMount()
expect(wrapper.state("vehicles")).toEqual([])
})
it("should change state when fetch is unresolved", async () => {
fetch = jest.fn().mockImplementationOnce(() => Promise.reject(
    new Error("fetch fail")
  ));
expect(wrapper.state("error")).toEqual("")
await wrapper.instance().componentDidMount()
expect(wrapper.state("error")).toBe("fetch fail")
})
})