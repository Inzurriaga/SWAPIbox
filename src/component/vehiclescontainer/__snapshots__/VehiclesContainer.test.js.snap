// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vehicles_container should match snapshot 1`] = `
<div
  className="vehicles-container"
/>
`;
